### PR TITLE
Use golang-1.17 as baseimage

### PR DESCRIPTION
**Pull request by kpouget**
/cc @jmencak

### CODE REVIEW

**Comment by jmencak** [comment]
Hmm, [failed to get CLI image](https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/27547/rehearse-27547-pull-ci-openshift-psap-ci-artifacts-release-4.8-nfd-images/1511243117112594432/build-log.txt) for release-4.8/nfd-images

**Comment by kpouget** [comment]
> Hmm, [failed to get CLI image](https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/27547/rehearse-27547-pull-ci-openshift-psap-ci-artifacts-release-4.8-nfd-images/1511243117112594432/build-log.txt) for release-4.8/nfd-images

that's a flake
```
  * could not run steps: step [release:latest] failed: failed to get CLI image: unable to find the 'cli' image in the provided release image: the pod ci-op-dwfr3nzl/release-images-latest-cli failed after 10s (failed containers: ): Evicted Pod The node had condition: [DiskPressure].  
```
but our Dockerfile already downloads its `oc` command, I guess `cli` NFD stanza could be removed ...

/retest

**Comment by kpouget** [comment]
> But it seems like a non-required test, so perhaps it is not fatal?

just FYI, the image test isn't "required" to merge the PR, but if it were an actual bug (wrong Dockerfile content), the image wouldn't build ever, so kind of required ;-)
besides, all the `*-images` tests (`nfd-images`, `4.8-images`, `4.7-images`, `ods-images`, `nto-images`) do the same thing in CI-artifacts (AFAIK), that's why in `ci-artifacts` PR testing, only one of them is enabled, and the others are never triggered.
Rehearsal blindly tests all of them :)

**Comment by kpouget** [comment]
@ArangoGutierrez NFD on 4.7 failed, because the sample [CR](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/27547/rehearse-27547-pull-ci-openshift-psap-ci-artifacts-master-4.7-gpu-operator-e2e/1511243116848353280/artifacts/gpu-operator-e2e/presubmit-operatorhub/artifacts/003__nfd_operator__deploy_from_operatorhub/003_operator_cr.json) doesn't contain the namespace, so `oc apply -f $FILE` used the implicite (but forbidden) namespace:
```
<command> oc apply -f /logs/artifacts/003__nfd_operator__deploy_from_operatorhub/003_operator_cr.json
<stderr> Error from server (NotFound): error when creating "/logs/artifacts/003__nfd_operator__deploy_from_operatorhub/003_operator_cr.json": namespaces "ci-op-ibzg15kh" not found
```
is that something expected, on NFD side? 
it used to work, so I'm surprised :/

**Comment by ArangoGutierrez** [comment]
> @ArangoGutierrez NFD on 4.7 failed, because the sample [CR](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/27547/rehearse-27547-pull-ci-openshift-psap-ci-artifacts-master-4.7-gpu-operator-e2e/1511243116848353280/artifacts/gpu-operator-e2e/presubmit-operatorhub/artifacts/003__nfd_operator__deploy_from_operatorhub/003_operator_cr.json) doesn't contain the namespace, so `oc apply -f $FILE` used the implicite (but forbidden) namespace:
> 
> ```
> <command> oc apply -f /logs/artifacts/003__nfd_operator__deploy_from_operatorhub/003_operator_cr.json
> <stderr> Error from server (NotFound): error when creating "/logs/artifacts/003__nfd_operator__deploy_from_operatorhub/003_operator_cr.json": namespaces "ci-op-ibzg15kh" not found
> ```
> 
> is that something expected, on NFD side? it used to work, so I'm surprised :/

Yes is expected. this was modified some time ago in NFD 
https://github.com/openshift/cluster-nfd-operator/pull/238

File: ci-operator/config/openshift-psap/ci-artifacts/openshift-psap-ci-artifacts-release-4.8__nfd.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ build_root:
   image_stream_tag:
     name: release
     namespace: openshift
-    tag: golang-1.16
+    tag: golang-1.17
```
Comment (ArangoGutierrez) [inline ci-operator/config/openshift-psap/ci-artifacts/openshift-psap-ci-artifacts-release-4.8__nfd.yaml:10]: No, the branch 4.8 should be tested with 1.16 
https://github.com/openshift/cluster-nfd-operator/blob/release-4.8/go.mod#L3

**Review by ArangoGutierrez — CHANGES_REQUESTED** [review]
I don't agree with a change for NFD, please revert

File: ci-operator/config/openshift-psap/ci-artifacts/openshift-psap-ci-artifacts-release-4.8__nfd.yaml (hunk 1 above)
Comment (kpouget) [reply ci-operator/config/openshift-psap/ci-artifacts/openshift-psap-ci-artifacts-release-4.8__nfd.yaml]: fixed

**Comment by jmencak** [comment]
> > is that something expected, on NFD side? it used to work, so I'm surprised :/
> 
> Yes is expected. this was modified some time ago in NFD [openshift/cluster-nfd-operator#238](https://github.com/openshift/cluster-nfd-operator/pull/238)

Right, so because 4.7 is in the Maintenance phase by August 24, 2022, we no longer care, right?

**Comment by kpouget** [comment]
I close this PR, I'm studying how to get a more recent base image in https://github.com/openshift/release/pull/29262.

**Pull request closed without merging**